### ISSUE TITLE
Plugin not showing on outlook addin list

### ISSUE THREAD

**Issue by mikesdamon** [wolfgangimig/ddaddin#4]
The installer runs successfully for me, 64-bit Windows 10 with 64-bit Outlook 2016, and a folder is created here: "C:\Program Files\WILUTIONS\DnD to HTML5 Addin for Microsoft Outlook" with 2 dlls, the HKLM registry keys listed in the wiki were created, but nothing shows on the add-in list in outlook. I tried adding manually, but when I select the 64 dll and hit OK, it just doesn't add to the list and no error pops up. I set the logging to debug, but no file was created. Restarted Outlook/Windows and no luck. My company uses a web based CRM and is attachment CRAZY so if I can get this working I'm thiniking we will be picking up lots of licenses...

**Comment by wolfgangimig** [comment]
Hi mikesdamon, 
1. Did you create your own setup? 
2. Did you restart Outlook after installation?
3. Please check whether the DLL is registered as a COM object:
   3.1. HKCU\WIDragDropAddin.class\CurVer default = WIDragDropAddin.class.1
   3.2. HKCU\WIDragDropAddin.class.1\CLSID={3967A635-CD8B-4960-966A-5BFEAF17273E}
   3.3. HKCU\CLSID{3967A635-CD8B-4960-966A-5BFEAF17273E}\InprocServer32 default=installationdirectory\ddaddin64.dll

Regards,
Wolfgang

**Comment by tigrente** [comment]
Hi Wolfgang - 
I'm having the same issue with Outlook 2016.  I am not using a custom installer. Things I tried:
1) Adding the above registry entries
2) Reinstalling, repairing, restarting Outlook and all office apps
3) Trying to add a COM plug in by browsing to ddaddin64.dll and selecting it.   It does not give any error messages, but is not added to the add-in list.  

Wilutions is not listed as a trusted publisher.  Could this be an issue? 

Thank you for your guidance,
Alex

**Comment by wolfgangimig** [comment]
Hi Alex,

currently, I do not have a certificate to sign the installer and the Addin. (I already applied for it). So the Trust Center options in Outlook have to enable unsigned Addins. I found this information related to this topic:
https://support.office.com/en-au/article/View-manage-and-install-add-ins-in-Office-programs-16278816-1948-4028-91e5-76dca5380f8d
In my Outlook 2016, there are only options for "Macro Settings" which have a box that let the settings also apply to addins. If the box is checked, DDAddin is not loaded - but still listed under File - Options - Addins - Inactive ... Addins. Is the Addin listed in your Outlook under active or inactive Addins?

Would you please check 3. again - the registry entries. 

Regards,
Wolfgang

**Comment by wolfgangimig** [comment]
A new - signed - version 1.0.11 of DDAddin is available now.
Please check whether it is installed properly now.

**Comment by insuranceshop** [comment]
Hi Wolfgang,

I have checked all 3 steps you listed above.

I ran both 32 bit and 64 bit installations on first a Windows 10 machine with Outlook 2016 and then a Win 7 with Outlook 2016 both with success, however, having the same issue others have reported above: I do not see the add-in in the add-ins panel.

When I checked the registry keys that you mentioned in step 3, I've noticed neither my Win7 box nor my Win10 box have generated them. Do you have any idea what the issue could be? In case there are things I can try or any new versions I can test I would be happy to... If we can get this to work I am interested in a lot of licenses as well :)

**Comment by wolfgangimig** [comment]
Thank you for your investigations.

During installation, a log file is written in the temp directory of the user, e.g. c:\Users\Wolfgang\AppData\Local\Temp\ddaddin.log. Please send this file to wilutions@gmail.com. 

Is there an entry for "DnD to HTML5 Addin for Microsoft Outlook" in the list of deactivated Add-ins? File - Options...
![image](https://cloud.githubusercontent.com/assets/3779094/12216198/927198ae-b6da-11e5-886c-6f55c62dc68a.png)

Are there non-default configurations in Outlook's "Trust Center"?
![image](https://cloud.githubusercontent.com/assets/3779094/12216230/273c451a-b6db-11e5-9902-d73fe9344d83.png)
![image](https://cloud.githubusercontent.com/assets/3779094/12216247/6722fbe2-b6db-11e5-84d9-525379a750ac.png)

Are there any group policy options entered regarding to Outlook as mentioned here:
https://technet.microsoft.com/en-us/library/ff657852.aspx#BKMK_COMAddIns

Thanks in advance for your effort and your patience.

**Comment by wolfgangimig** [comment]
Hi Fersu,
The provided log file does not report any errors. Everything works fine during installation. The reg-keys mentioned above are correctly created under HKEY_CLASSES_ROOT. It was my mistake to give the wrong part so you couldn't find them, sorry. 
Your screenshots from Outlook options look normal.

Please go through the following questions:
1 The installation procedure detects Outlook as a 32bit application. Is this correct?
2 Are there any warnings or errors from Outlook in the Windows event log viewer?
3 In order to check the registration of DDAddin, copy the following lines into a VBS file, e.g. c:\test.vbs

```
set addin = CreateObject("WIDragDropAddin.class")
MsgBox "Addin loaded"
Set oShell = CreateObject("WScript.Shell")
LoadBehavior = oShell.RegRead("HKEY_LOCAL_MACHINE\SOFTWARE\WOW6432Node\Microsoft\Office\Outlook\Addins\WIDragDropAddin.class\LoadBehavior")
MsgBox "LoadBehavior=" & LoadBehavior
```

then execute in a command window:

```
c:\windows\syswow64\wscript.exe c:\test.vbs
```

A message box should appear with text "Addin loaded". Then, another box should print "LoadBehavior=3".

Kind Regards,
Wolfgang

**Comment by wolfgangimig** [comment]
insuranceshop has confirmed that the issue is solved with version 1.0.12.
The installer did not recognize that Outlook 2016 is installed as 64bit version.
Many thanks to insuranceshop for your valuable help.

**Comment by tigrente** [comment]
Hi Wolfgang - Thank you very much for responding.  I’m currently on a plane to China, but I will attempt this later this weekend.Best regards,AlexOn Jan 5, 2016, at 12:33 PM, Wolfgang Imig notifications@github.com wrote:Hi Alex,currently, I do not have a certificate to sign the installer and the Addin. (I already applied for it). So the Trust Center options in Outlook have to enable unsigned Addins. I found this information related to this topic:
https://support.office.com/en-au/article/View-manage-and-install-add-ins-in-Office-programs-16278816-1948-4028-91e5-76dca5380f8d
In my Outlook 2016, there are only options for "Macro Settings" which have a box that let the settings also apply to addins. If the box is checked, DDAddin is not loaded - but still listed under File - Options - Addins - Inactive ... Addins. Is the Addin listed in your Outlook under active or inactive Addins?Would you please check 3. again - the registry entries. Regards,
Wolfgang—Reply to this email directly or view it on GitHub.
